hub2vu/homework_vector_db_2.py

```
@@ -0,0 +1,190 @@
+    (토큰 상한 여유 있게. 필요시 조정)
+    """
+    s = re.sub(r"[ \t]+", " ", s)
+    s = re.sub(r"\n{3,}", "\n\n", s)
+    if len(s) > max_chars:
+        s = s[:max_chars]
+    return s
+
+def chunk_text_generator(text: str, chunk_size: int = 400, chunk_overlap: int = 50) -> Iterator[str]:
+    """ 제너레이터로 청크 생성 (메모리 안전) """
+    if chunk_overlap >= chunk_size:
+        raise ValueError("chunk_overlap은 chunk_size보다 작아야 합니다.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)